Draft: ciba-redirect-refactor (#103)
vippsas/vipps-login-api: vipps-login-api.md

    Authorization: Basic asdkjhasdjhsad=
    Content-Type: application/x-www-form-urlencoded

    requested_flow=login_to_webpage&scope=openid name address&login_hint=urn:mobilenumber:{mobileNumber}&redirect_uri=https://merchantwebpage.com/callback
    ```

>     Example response, the `auth_req_id` can later be used to connect this login to a token response since the ID token should contain the same `auth_req_id` value.

Reviewer: ```suggestion
    Example response, the `auth_req_id` should be used to connect this login to a token response since the ID token should contain the same `auth_req_id` value.
```

      "interval": 5
    }
    ```

> 2. The user confirms the login and is then redirected to the `redirect_uri` passed in the initial request 1. The redirect will contain `code` and `state` parameters: `{redirect_uri}?code={code}`.

Author: ```suggestion
2. The user confirms the login and is then redirected to the `redirect_uri` passed in the initial request 1. The redirect will contain a `code`: `{redirect_uri}?code={code}`.
```

      "interval": 5
    }
    ```
2. The user confirms the login and is then redirected to the `redirect_uri` passed in the initial request 1. The redirect will contain a `code`: `{redirect_uri}?code={code}`.

> 3. The merchant uses the code-parameter to obtain the login token. POST `{token_endpoint}` with `code={code}`, `grant_type=urn:vipps:params:gran-type:ciba-redirect` in the `application/x-www-form-urlencoded-body`. This returns a ID token and an access token that can be used to fetch userinfo. The ID token is an JWS that must be validated, see https://github.com/vippsas/vipps-login-api/blob/master/vipps-login-api.md#id-token. The merchant **must validate** that it contains a `auth_req_id` they have previously received from step 2.

Reviewer: ```suggestion
3. The merchant uses the code-parameter to obtain the login token. POST `{token_endpoint}` with `code={code}`, `grant_type=urn:vipps:params:grant-type:ciba-redirect` in the `application/x-www-form-urlencoded-body`. This returns an ID token and an access token that can be used to fetch userinfo. The ID token is a JWS that must be validated, see https://github.com/vippsas/vipps-login-api/blob/master/vipps-login-api.md#id-token. The merchant **must validate** that it contains the `auth_req_id` they have previously received from step 2.
```

> 3. The merchant uses the code-parameter to obtain the login token. POST `{token_endpoint}` with `code={code}`, `grant_type=urn:vipps:params:gran-type:ciba-redirect` in the `application/x-www-form-urlencoded-body`. This returns a ID token and an access token that can be used to fetch userinfo. The ID token is an JWS that must be validated, see https://github.com/vippsas/vipps-login-api/blob/master/vipps-login-api.md#id-token. The merchant **must validate** that it contains a `auth_req_id` they have previously received from step 2.

Reviewer 2: Shouldn't it be .."the id token is a JWT with a JWS that must be validated"..

Author: Pretty sure JWS is the correct term for the whole payload even according to the standard itself.

    Example request (the real payload will likely look different because of encoding):
    ```

>     POST https://api.vipps.no

Reviewer: ```suggestion
    POST https://api.vipps.no/access-management-1.0/access/oauth2/token
```

    Authorization: Basic sadlksadkjasjdaksd
    Content-Type: application/x-www-form-urlencoded

>     code=some-valid-code&grant_type=urn:vipps:params:gran-type:ciba-redirect

Reviewer: ```suggestion
    code=some-valid-code&grant_type=urn:vipps:params:grant-type:ciba-redirect
```
